Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -146,12 +146,12 @@ public class ChatChannelController: DataController, DelegateCallable, DataStoreP
 
     /// A boolean value indicating if it should send typing events.
     /// It is `true` if the channel typing events are enabled as well as the user privacy settings.
-    internal var shouldSendTypingEvents: Bool {
-        /// Ignore if user typing indicators privacy settings are disabled. By default, they are enabled.
-        let currentUserPrivacySettings = client.sharedCurrentUserController.currentUser?.privacySettings
-        let isTypingIndicatorsForCurrentUserEnabled = currentUserPrivacySettings?.typingIndicators?.enabled ?? true
-        let isChannelTypingEventsEnabled = channel?.canSendTypingEvents ?? true
-        return isTypingIndicatorsForCurrentUserEnabled && isChannelTypingEventsEnabled
+    func shouldSendTypingEvents(completion: @escaping (Bool) -> Void) {
+        guard channel?.canSendTypingEvents ?? true else {
+            completion(false)
+            return
+        }
+        eventSender.currentUserTypingEventsState(completion: completion)
```

Comment:
Could we do this instead? 🤔 The DataStore has access to the current user, so it is probably the best approach in this scenario? It will simplify the code as well
```suggestion
        let currentUserPrivacySettings = dataStore.currentUser().privacySettings
        let isTypingIndicatorsForCurrentUserEnabled = currentUserPrivacySettings?.typingIndicators?.enabled ?? true
        let isChannelTypingEventsEnabled = channel?.canSendTypingEvents ?? true
        return isTypingIndicatorsForCurrentUserEnabled && isChannelTypingEventsEnabled
```

Reply:
It is going to be expensive because `dataStore.currentUser()` fetches the DTO, then converts it to the full immutable type.

Reply:
I see...